utils/ansible/ftest/library/daos_hugepages.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+__copyright__ = """
+(C) Copyright 2018-2022 Intel Corporation.
+
+SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '0.1',
+    'status': ['preview'],
+    'supported_by': 'Intel'
+}
+
+
+DOCUMENTATION = '''
+---
+module: daos_hugepages
+
+short_description: Configure Linux Huge Pages
+
+version_added: "0.1"
+
+description:
+    - Check if the Linux Huge Pages feature is enabled
+    - Check the value of the Huge Pages
+    - Set the value of the Huge Pages if needed
+
+options:
+    size:
+        description:
+            - Number of huge pages to allocate
+        type: int
+        default: 4096
+
+author:
+    - DAOS group <daos@daos.groups.io>
+'''
+
+EXAMPLES = '''
+- name: Set the Linux Huge Pages size
+  daos_hugepages
+'''
+
+RETURN = '''
+elapsed:
+     description: the number of seconds that elapsed while checking and setting the huge pages size
+     returned: always
+     type: int
+     sample: 3
+'''
+
+import os
+import re
+import datetime
+import subprocess
+
+from ansible.module_utils.basic import AnsibleModule
+
+__HUGEPAGES_STATE_PATTERN__ = re.compile(r"(\[madvise\])|(\[always\])")
+__HUGEPAGES_SYSCTL_PATTERN = re.compile(r"^vm\.nr_hugepages\s*=\s*(?P<size>\d+)$")
+
+def is_huge_pages_enabled():
+    with open(r'/sys/kernel/mm/transparent_hugepage/enabled', 'r') as fd:
+        line = fd.read()
+        return __HUGEPAGES_STATE_PATTERN__.match(line)
+
+def main():
```

Comment:
What is this file for? This PR is for building DAOS and running functional tests, but dealing with hugepages seems random.

Reply:
This PR is also (and mainly) intended to prepare the different nodes on which we want to run the functional tests.
This python module is intended to check and update (if needed) the huge pages configuration of the targeted node.
To better understand this PR, reading this file could probably help:
https://github.com/daos-stack/daos/blob/5bc4e2530905363ea659b6458a5be770b41bdeb7/utils/ansible/ftest/README.md

Reply:
At the end, the aims of this work is to help developers such as me to develop and validate functional tests.